### PR TITLE
systemtests: make them usably with Windows binaries on Linux using Wine

### PR DESCRIPTION
**Pull request by joergsteffens**
### Thank you for contributing to the Bareos Project!

Make the Windows binaries testable via systemtests on Linux using Wine.

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

##### General
- [ ] PR name is meaningful
- [ ] Purpose of the PR is understood
- [ ] Commit descriptions are understandable and well formatted
- [ ] Check backport line
- [ ] Is the PR title usable as CHANGELOG entry?
- [ ] Separate commit for CHANGELOG.md ("update CHANGELOG.md"). The PR number is correct.
- [ ] Has https://git.bareos.com/bareos/CD-buildimages/pulls/12 been merged?
- [ ] Has https://git.bareos.com/bareos/CD/pulls/53/ been merged?

##### Source code quality

- [ ] Source code changes are understandable
- [ ] Variable and function names are meaningful
- [ ] Code comments are correct (logically and spelling)
- [ ] Required documentation changes are present and part of the PR
- [ ] `bareos-check-sources --since-merge` does not report any problems

##### Tests

- [ ] Decision taken that a test is required (if not, then remove this paragraph)
- [ ] The choice of the type of test (unit test or systemtest) is reasonable
- [ ] Testname matches exactly what is being tested
- [ ] On a fail, output of the test leads quickly to the origin of the fault

### CODE REVIEW

File: systemtests/CMakeLists.txt [hunk 1]
```diff
@@ -74,34 +99,38 @@ message(STATUS "POSTGRES_BIN_PATH is ${POSTGRES_BIN_PATH}")
 
 find_program(PERL perl)
 
+set(SYSTEMTESTS_USE_WINE
+    ""
+    CACHE STRING "systemtests do use WINE to run Bareos"
+)
```
Comment (arogge) [inline systemtests/CMakeLists.txt:105]: If you want a boolean cache variable, you can use option.
```suggestion
option(SYSTEMTESTS_USE_WINE
       "systemtests do use WINE to run Bareos"
       OFF
)
```

Comment (joergsteffens) [reply systemtests/CMakeLists.txt]: Applied.

**Comment by joergsteffens** [comment]
> That this PR alters more thant 600 `.in` files just to add a prefix to an exisiting variable is not a good solution. Probably the variables need to be prefixed themselves by cmake instead.

The problem is, that we need this variable in 2 variants:
* Unix path format (for inside cmake and shell scripts)
* Windows path format. When used as Bareos configuration, as the Daemon do run in Windows/Wine. In Bareos we can use the same path, but must prefix it with `@wine_drive_prefix@`.

If using `@wine_drive_prefix@` in the Bareos configuration files is not an option, I still have to create a second variable for every path variable used in the Bareos configuration. Example:

Instead of using
* `@archivedir@` and
* `@wine_drive_prefix@@archivedir@`
I could create a second variable to be used in the Bareos configuration:
* `@archivedir@` and
* `@bcfg_archivedir@` (= `@wine_drive_prefix@@archivedir@`)

Still, the same amount if `*.in` files must be modified. Do you think, the second variant is better? Would you agree to it?

**Comment by joergsteffens** [comment]
> That this PR alters more thant 600 `.in` files just to add a prefix to an exisiting variable is not a good solution. Probably the variables need to be prefixed themselves by cmake instead.

The problem is, that we need this variable in 2 variants:
* Unix path format (for inside cmake and shell scripts)
* Windows path format. When used as Bareos configuration, as the Daemon do run in Windows/Wine. In Bareos we can use the same path, but must prefix it with `@wine_drive_prefix@`.

If using `@wine_drive_prefix@` in the Bareos configuration files is not an option, I still have to create a second variable for every path variable used in the Bareos configuration. Example:

Instead of using
* `@archivedir@` and
* `@wine_drive_prefix@@archivedir@`
I could create a second variable to be used in the Bareos configuration:
* `@archivedir@` and
* `@bcfg_archivedir@` (= `@wine_drive_prefix@@archivedir@`)

Still, the same amount if `*.in` files must be modified. Do you think, the second variant is better? Would you agree to it?

**Comment by bruno-at-bareos** [comment]
Closing, a try will be make to run test in native builds.

**Pull request closed without merging**